File: lab1/e.py
import math
a = list(map(int, input().split()))
b = list(map(int, input().split()))
cnt = 0
bound = math.pow(10, 6)
while True:
    if cnt >= bound:
        print('blin nichya')
        break
    elif len(a) == 0:
        print("Nursik", cnt)
        break
    elif len(b) == 0:
        print("Boris", cnt)
        break
    elif a[0] > b[0]:
        tmp = a[0]
        a.pop(0)
        a.append(tmp)
        a.append(b[0])
        b.pop(0)
    elif len(b) != 0 and b[0] > a[0]:
        tmp = b[0]
        b.pop(0)
        b.append(a[0])
        b.append(tmp)
        a.pop(0)
    cnt += 1

    

File: lab1/e.cpp
#include <bits/stdc++.h>
using namespace std;
void input(queue<int> &q){
    for(int i = 0; i < 5; i++){
        int a;
        cin >> a;
        if(a == 0)q.push(0);
        else q.push(a);
    }
}
int main() {
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);
    queue<int> q1, q2;
    input(q1);
    input(q2);
    int cnt = 0;
    while(1){
        if(cnt >= pow(10, 6)){
            cout << "blin nichya";
            break;
        }
        else if(q1.empty()){
            cout << "Nursik " << cnt; 
            break;
        }
        else if(q2.empty()){
            cout << "Boris " << cnt; 
            break;
        }
        else if(q1.front() >= q2.front()){
            if(q1.front() == 9 && q2.front() == 0){
                int tmp = q2.front();
            q2.pop();
            q2.push(q1.front());
            q2.push(tmp);
            q1.pop();
            }
            else{
                int tmp = q1.front();
                q1.pop();
                q1.push(tmp);
                q1.push(q2.front());
                q2.pop();
            }
        }
        else if(!q2.empty() && q1.front() < q2.front()){
            if(q1.front() == 0 && q2.front() == 9){
                int tmp = q1.front();
                q1.pop();
                q1.push(tmp);
                q1.push(q2.front());
                q2.pop();
            }
            else{
                int tmp = q2.front();
                q2.pop();
                q2.push(q1.front());
                q2.push(tmp);
                q1.pop();
            }
        }
        cnt++;
    }
    return 0;
}